common/src/addressing.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::digest::Digest;
+use crypto::sha2::Sha256;
+
+const FAMILY_NAMESPACE: &str = "bond namespace";
+const ORGANIZATION: &str = "organization";
+const PARTICIPANT: &str = "participant";
+const BOND: &str = "bond";
+const HOLDING: &str = "holding";
+const SETTLEMENT: &str = "settlement";
+const RECEIPT: &str = "receipt";
+const ORDER: &str = "order";
+const QUOTE: &str = "quote";
+
+pub fn get_prefix(object: &str) -> String {
+    let mut sha = Sha256::new();
+    sha.input_str(object);
+    sha.result_str()[..6].to_string()
+}
+
+pub fn hash(to_hash: &str, num: usize) -> String {
+    let mut sha = Sha256::new();
+    sha.input_str(to_hash);
+    let temp = sha.result_str().to_string();
+    let hash = match temp.get(..num) {
```

Comment:
I'm also confused on why you need that match statement on `hash` but not on `get_prefix`. It seems to me you could refactor `hash` to look more like `get_prefix`, which is a little cleaner?

Reply:
Yeah, I initially had a function specifically for the "bond_namespace" prefix, then realized that the object type was also the same size so just consolidated that function for the prefixes... I'll consolidate those functions!